internal/clusterconfig/aws_client.go

```diff
@@ -0,0 +1,31 @@
+package clusterconfig
+
+import (
+	"net/http"
+	"time"
+)
+
+const awsMetadataHost = "http://169.254.169.254"
+
+// TODO: get rid of Interface in the name bellow
+
+type AwsClientInterface interface {
+	IsAws() bool
+}
+type AwsClient struct {
+	*http.Client
+}
+
+func NewAwsClient() *AwsClient {
+	return &AwsClient{
+		&http.Client{Timeout: 5 * time.Second},
```

Comment:
The timeout is rather long, because on all non-AWS environments this will run into a timeout. So I think a rather short timeout like 1s or 2s would be better.